test/credentials_test.dart

```diff
@@ -61,36 +61,99 @@ Future<void> main([List<String>? args]) async {
     expect(user, isNotNull);
   });
 
+  baasTest('Email/Password - register user auto confirm', (configuration) async {
+    final app = App(configuration);
+    final authProvider = EmailPasswordAuthProvider(app);
+    String username = "${generateRandomString(5)}@bar.com";
+    String password = "SWV23R#@T#VFQDV";
+    // For application with name 'auto' and with confirmationType = 'autoConfirm'
+    // all the usernames are automatically confirmed.
+    await authProvider.registerUser(username, password);
```

Comment:
Can we add a test that we can now login with the credentials? FYI, there's [a bug in Cloud](https://jira.mongodb.org/browse/REALMC-12253) that means that you may have to retry the login since there's a small delay between registering and the autoconfirmation kicking in.

Reply:
Done with retryLogin function